src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt

```diff
@@ -3300,7 +3300,7 @@ Expression LikeExpression(Expression leftExpression) #LikeExpression:
     Expression escape;
 }
 {
-    [<K_NOT> { result.setNot(true); } ] ( <K_LIKE> | <K_ILIKE> { result.setCaseInsensitive(true); } ) rightExpression=SimpleExpression()
+    [<K_NOT> { result.setNot(true); } ] ( <K_LIKE> | <K_ILIKE> | <K_RLIKE> { result.setCaseInsensitive(true); } ) rightExpression=SimpleExpression()
```

Comment:
Unfortunately, you break the Logic here:
1) For `<K_LIKE>` only, `caseInsensitive:=false` and
2) For `<K_ILIKE>` only,   `caseInsensitive:=false` 

You set `caseInsensitive` for `<K_ILKE>` instead.